From 00da41cb15bbe3d552eec36d2814081396b5bc86 Mon Sep 17 00:00:00 2001
From: Bharat Pawar <bpawar@codeaurora.org>
Date: Mon, 7 Jun 2021 18:27:45 +0530
Subject: [PATCH] wlan: Fix compile issue for wpa_supplicant_8_lib

send_and_recv_msgs function is guarded by
CONFIG_ANDROID_S macro as the newer API has
introduced two additional parameters with android S.

CRs-Fixed: 3045269
Change-Id: I251396cc554d6a43e546e2a88e6740bf4fd3a357
---
 qcwcn/wpa_supplicant_8_lib/Android.mk           |  4 ++++
 qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211.c | 16 ++++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/qcwcn/wpa_supplicant_8_lib/Android.mk b/qcwcn/wpa_supplicant_8_lib/Android.mk
index e33f9cc..1d21546 100644
--- a/qcwcn/wpa_supplicant_8_lib/Android.mk
+++ b/qcwcn/wpa_supplicant_8_lib/Android.mk
@@ -54,6 +54,10 @@ ifdef CONFIG_P2P
 L_CFLAGS += -DCONFIG_P2P
 endif
 
+ifneq ( ,$(filter S 12, $(PLATFORM_VERSION)))
+L_CFLAGS += -DCONFIG_ANDROID_S
+endif
+
 L_CFLAGS += -Werror
 
 ########################
diff --git a/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211.c b/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211.c
index 48538ff..949ba91 100644
--- a/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211.c
+++ b/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211.c
@@ -95,7 +95,11 @@ static int wpa_driver_cmd_set_ani_level(struct i802_bss *bss, int mode, int ofdm
 		}
 	}
 	nla_nest_end(msg, params);
+#ifndef CONFIG_ANDROID_S
 	ret = send_and_recv_msgs(drv, msg, NULL, NULL);
+#else
+	ret = send_and_recv_msgs(drv, msg, NULL, NULL, NULL, NULL);
+#endif
 	if (!ret)
 		return 0;
 	wpa_printf(MSG_ERROR, "%s: Failed set_ani_level, ofdmlvl=%d, ret=%d",
@@ -152,7 +156,11 @@ static int wpa_driver_cmd_set_congestion_report(struct i802_bss *bss, char *cmd)
 		}
 	}
 	nla_nest_end(msg, params);
+#ifndef CONFIG_ANDROID_S
 	ret = send_and_recv_msgs(drv, msg, NULL, NULL);
+#else
+	ret = send_and_recv_msgs(drv, msg, NULL, NULL, NULL, NULL);
+#endif
 	wpa_printf(MSG_INFO, "%s: set congestion report: enable=%d, threshold=%d,"
 			"interval=%d", __FUNCTION__, enable, threshold, interval);
 	if (!ret)
@@ -190,7 +198,11 @@ static int wpa_driver_cmd_set_tx_power(struct i802_bss *bss, char *cmd)
 		return -ENOBUFS;
 	}
 
+#ifndef CONFIG_ANDROID_S
 	ret = send_and_recv_msgs(drv, msg, NULL, NULL);
+#else
+	ret = send_and_recv_msgs(drv, msg, NULL, NULL, NULL, NULL);
+#endif
 	if (!ret)
 		return 0;
 
@@ -1986,7 +1998,11 @@ static int wpa_driver_cmd_get_thermal_info(struct i802_bss *bss, int *result, in
 	}
 
 	nla_nest_end(msg, params);
+#ifndef CONFIG_ANDROID_S
 	ret = send_and_recv_msgs(drv, msg, thermal_info_handler, result);
+#else
+	ret = send_and_recv_msgs(drv, msg, thermal_info_handler, result, NULL, NULL);
+#endif
 	if (!ret)
 		return 0;
 	wpa_printf(MSG_ERROR, "%s: Failed get thermal info, ret=%d(%s)",
-- 
2.34.0

